### ISSUE TITLE
Converting models to TFLite is not deterministic

### ISSUE THREAD

**Issue by nitsanh** [tensorflow/tensorflow#55512]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Mac 11.5.2
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.7.1
- Python version: 3.8.8
- GPU model and memory: AMD Radeon Pro 5300M 4 GB, Intel UHD Graphics 630 1536 M

**Describe the current behavior**
I have a keras model that I convert into a tflite model using `tf.lite.TFLiteConverter.from_keras_model`.
I then save the converted tflite model. I noticed that the output of the conversion is not bit-exact. That means if I save the same converted tflite model twice, the two files may have different hashes.

**Describe the expected behavior**
I would expect the output tflite model file to always have the same hash. I need this so I will be able to know if the model changed.

**Standalone code to reproduce the issue**
This behavior doesn't reproduce every time, but it happens pretty often

```
import os
import tensorflow as tf

def create_model():
    inputs = tf.keras.Input(shape=[5, 5, 3])
    x = tf.keras.layers.Conv2D(32, 3, strides=2, padding="same")(inputs)
    x = tf.keras.layers.BatchNormalization()(x)
    model = tf.keras.Model(inputs, x)
    return model


model = create_model()

converter = tf.lite.TFLiteConverter.from_keras_model(model)
converter.experimental_new_converter = True
converter.optimizations = [tf.lite.Optimize.DEFAULT]
converter.target_spec.supported_types = [tf.float16]

tflite_model1 = converter.convert()

tflite_model2 = converter.convert()

with open("model1.tflite", "wb") as f:
    f.write(tflite_model1)

with open("model2.tflite", "wb") as f:
    f.write(tflite_model2)

print(os.system("diff model1.tflite model2.tflite"))


converter = tf.lite.TFLiteConverter.from_keras_model(model)
converter.experimental_new_converter = True
converter.optimizations = [tf.lite.Optimize.DEFAULT]
converter.target_spec.supported_types = [tf.float16]

tflite_model3 = converter.convert()

with open("model3.tflite", "wb") as f:
    f.write(tflite_model3)

print(os.system("diff model1.tflite model3.tflite"))
```

**Comment by mohantym** [comment]
Hi @nitsanh ! I am able to get deterministic result  by using tf.random.set_seed() . Attaching [gist ](https://colab.sandbox.google.com/gist/mohantym/04440e592dc4b95534f274be7fb1e8cc/git_55512.ipynb#scrollTo=4Tg-ibeGOaTQ)for reference . Thanks!

**Comment by nitsanh** [comment]
Hi @mohantym, thanks for the quick reply! I ran your gist again and received non-deterministic output when using a seed and version 2.8.0 [here](https://colab.research.google.com/gist/nitsanh/dc970085a2bd9829f13e1086e0505458/git_55512.ipynb)

**Comment by mohantym** [comment]
@nitsanh ! I used[ tf.config.experimental.enable_op_determinism ](https://www.tensorflow.org/api_docs/python/tf/config/experimental/enable_op_determinism)to get deterministic result . It did replicate in [2.8](https://colab.sandbox.google.com/gist/mohantym/a2786086d2cab11291a30b35bb4f3f31/git_55512.ipynb) though , but seems to be resolved in the [nightly](https://colab.sandbox.google.com/gist/mohantym/98b552b6f6aec62686c7602380f73674/git_55512.ipynb#scrollTo=rVnvTAzfXam6) version.  Can you let us know from your end too? Thanks!

**Comment by nitsanh** [comment]
Hi @mohantym, it seems to be deterministic on my end as well using the nightly version. Thanks!
